cmake/modules/buildtools/FindClangFormat.cmake

```diff
@@ -8,4 +8,11 @@ find_program(CLANG_FORMAT_EXECUTABLE clang-format)
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(ClangFormat REQUIRED_VARS CLANG_FORMAT_EXECUTABLE)
 
+execute_process(COMMAND clang-format --version ERROR_QUIET OUTPUT_VARIABLE CLANG_FORMAT_VERSION_OUTPUT)
+string(REGEX MATCH "([0-9]+)\\..*" CLANG_FORMAT_VERSION ${CLANG_FORMAT_VERSION_OUTPUT})
+if(CLANG_FORMAT_VERSION VERSION_LESS_EQUAL "9.0.0")
+    message(WARNING "clang-format must be at least version 9.0 and the version installed is ${CLANG_FORMAT_VERSION}")
+    set(CLANG_FORMAT_FOUND OFF)
+endif()
+
```

Comment:
Can i suggest the following

```
#.rst:
# FindClangFormat
# ----------
# Finds clang-format

find_program(CLANG_FORMAT_EXECUTABLE clang-format)

include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(ClangFormat REQUIRED_VARS CLANG_FORMAT_EXECUTABLE)

if(CLANGFORMAT_FOUND)

  set(CLANGFORMAT_MINVERSION "9.0")

  execute_process(COMMAND clang-format --version ERROR_QUIET OUTPUT_VARIABLE CLANG_FORMAT_VERSION_OUTPUT)
  string(REGEX MATCH "([0-9]+)\\..*" CLANG_FORMAT_VERSION ${CLANG_FORMAT_VERSION_OUTPUT})
  if(CLANG_FORMAT_VERSION VERSION_LESS_EQUAL ${CLANGFORMAT_MINVERSION})
    message(WARNING "clang-format must be at least version ${CLANGFORMAT_MINVERSION}. The version found is ${CLANG_FORMAT_VERSION}")
    set(CLANGFORMAT_FOUND OFF CACHE BOOL "" FORCE)
    set(CLANG_FORMAT_EXECUTABLE "CLANG_FORMAT_EXECUTABLE-NOTFOUND" CACHE STRING "" FORCE)
  endif()
endif()

mark_as_advanced(CLANG_FORMAT_EXECUTABLE)

```

We only do the version check if clang-format is found. We extract the version number to s single variable (for if/when it needs to be updated).
It also sets the cache variable for CLANGFORMAT_FOUND and CLANG_FORMAT_EXECUTABLE to false/notfound